src/main/java/edu/kis/powp/jobs2d/drivers/TransformingDriver.java

```diff
@@ -0,0 +1,87 @@
+package edu.kis.powp.jobs2d.drivers;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+/**
+ * Decorator for Job2dDriver that applies transformations before forwarding the instructions.
+ */
+public class TransformingDriver implements Job2dDriver {
+    private final Job2dDriver driver;
+    private double scaleX;
+    private double scaleY;
+    private double rotationAngle;
+    private boolean isFlipped;
+    private int offsetX;
+    private int offsetY;

```

Comment:
Try refactoring the proposed solution to a more OCP (open-closed principle) approach. Adding a new transformation is now possible by:
1. modifying this class - not 'closed' for modification,
2. adding a new class similar to TransformingDriver - would make an unclear division of responsibility between those classes.